core/HoistAuthModel.ts

```diff
@@ -0,0 +1,87 @@
+/*
+ * This file belongs to Hoist, an application development toolkit
+ * developed by Extremely Heavy Industries (www.xh.io | info@xh.io)
+ *
+ * Copyright © 2024 Extremely Heavy Industries Inc.
+ */
+import {HoistModel, PlainObject, XH} from './';
+
+/**
+ *  Base class for managing authentication lifecycle.
+ *
+ *  Hoist will consult this model early in  the initialization sequence, prior to
+ *  full Hoist initialization. This means that several core services (identity, configs, prefs)
+ *  will *not* be available, but it provides the app a hook to
+ *  do early service initialization or other work to support flows such as OAuth.
+ *
+ *  The base implementation of this class is minimal and would be adequate only for fully
+ *  transparent SSO based solutions such as NTLM.  Applications wishing to provide custom
```

Comment:
This choice of words here, "fully transparent", doesn't clarify anything for me.  What would a "fully opaque" SSO based solution be? Or "half transparent"?  

I guess "transparent" is being used as a metaphor or an analogy for something, but I can't tell what.  Why is NTLM transparent?  Will all developers reading this understand why NTLM is transparent (except me)?

Reply:
sorry or the hand waving